### PR TITLE
Fix missing postings in Merge and Intersect

### PR DESCRIPTION
**Pull request by Gouthamve**
Before we were moving the postings list everytime we create a new
chained `intersectPostings`. That was causing some postings to be
skipped. This test fails on the older version.

Signed-off-by: Goutham Veeramachaneni <cs14btech11014@iith.ac.in>

---
This change is [Reviewable](https://reviewable.io/reviews/prometheus/tsdb/77)

### CODE REVIEW

**Comment by Gouthamve** [comment]
This PR fixes https://github.com/prometheus/prometheus/issues/2616

@fabxc The tests pass in my local and travis failure is not related to this PR. This is ready for review.

@telendt I rolled back always advancing on Seek as it was causing issues, please take a look at the explanation here: https://github.com/prometheus/tsdb/pull/77/commits/101c751add08c0beb5ee3bdc10c47c81a1776ca3 Would be good to have your review.

File: postings_test.go [hunk 1]
```diff
@@ -78,23 +78,35 @@ func TestIntersect(t *testing.T) {
...
-			c:   []uint32{1, 2, 5, 6, 7, 8, 1001, 1200},
+			p: [][]uint32{
+				{1, 2, 3, 4, 5, 6, 1000, 1001},
+				{2, 4, 5, 6, 7, 8, 999, 1001},
+				{1, 2, 5, 6, 7, 8, 1001, 1200},
+			},
 			res: []uint32{2, 5, 6, 1001},
 		},
+		{
+			p: [][]uint32{
```
Comment (fabxc) [inline postings_test.go]: I suppose that's the test verifying the bug is gone? Probably good to add a comment here which edge case this seemingly trivial test is checking.

Comment (Gouthamve) [inline postings_test.go]: Done
